hub/core/dataset/dataset.py

```diff
@@ -239,26 +239,31 @@ def __getitem__(
         is_iteration: bool = False,
     ):
         if isinstance(item, str):
-            fullpath = posixpath.join(self.group_index, item)
-            tensor = self._get_tensor_from_root(fullpath)
-            if tensor is not None:
-                return tensor[self.index]
-            elif self._has_group_in_root(fullpath):
-                return self.__class__(
-                    storage=self.storage,
-                    index=self.index,
-                    group_index=posixpath.join(self.group_index, item),
-                    read_only=self.read_only,
-                    token=self._token,
-                    verbose=False,
-                    version_state=self.version_state,
-                    path=self.path,
-                )
-            elif "/" in item:
-                splt = posixpath.split(item)
-                return self[splt[0]][splt[1]]
+            index=self.index
+            length=len(self)
+            if(index > length):
```

Comment:
`index` is a `hub.core.index.Index` object. Regardless, this check should be done in `tensor.py` as a dataset can have tensors of different lengths.